objects/src/accounts/component/template/mod.rs

```diff
@@ -277,6 +294,38 @@ impl AccountComponentMetadata {
             }
         }
 
+        self.validate_map_keys()?;
+
+        Ok(())
+    }
+
+    /// Validates map keys.
```

Comment:
```suggestion
    /// Validates map keys by checking for duplicates.
    ///
    /// This can only check keys given as immediate values (i.e. not placeholders) since placeholders are replaced at a later step at which point another duplicate check is done.
```